common/src/main/java/org/apache/cassandra/sidecar/common/CassandraAdapterDelegate.java

```diff
@@ -133,45 +133,47 @@ private void healthCheckInternal()
         if (activeSession == null)
         {
             logger.info("No local CQL session is available. Cassandra is down presumably.");
-            isUp = false;
+            nodeSettings = null;
             return;
         }
 
         maybeRegisterHostListener(activeSession);
 
         try
         {
-            String version = activeSession.execute("select release_version from system.local")
-                                          .one()
-                                          .getString("release_version");
-            isUp = true;
+            Row oneResult = activeSession.execute("select release_version, partitioner from system.local")
+                                   .one();
+            NodeSettings nodeSettings = new NodeSettings(oneResult.getString("release_version"),
+                                                   oneResult.getString("partitioner"));
+            this.nodeSettings = nodeSettings;
             // this might swap the adapter out
-            SimpleCassandraVersion newVersion = SimpleCassandraVersion.create(version);
+            SimpleCassandraVersion newVersion = SimpleCassandraVersion.create(nodeSettings.releaseVersion());
             if (!newVersion.equals(currentVersion))
             {
                 currentVersion = newVersion;
-                adapter = versionProvider.getCassandra(version).create(cqlSession);
+                adapter = versionProvider.getCassandra(nodeSettings.releaseVersion()).create(cqlSession);
                 logger.info("Cassandra version change detected. New adapter loaded: {}", adapter);
             }
-            logger.debug("Cassandra version {}", version);
+            logger.debug("Cassandra version {}", nodeSettings.releaseVersion());
         }
         catch (NoHostAvailableException e)
         {
             logger.error("Unexpected error connecting to Cassandra instance.", e);
             // The cassandra node is down.
             // Unregister the host listener and nullify the session in order to get a new object.
-            isUp = false;
+            nodeSettings = null;
             maybeUnregisterHostListener(activeSession);
+            session.close();
```

Comment:
we probably need to wrap this in a try/catch block , errors coming from the session should be safely ignored. And we probably want `activeSession` here instead of `session`